lib/display.rb

```diff
@@ -39,57 +79,92 @@ def valid_column(size)
     end
   end
 
-  def user_input_row
-    @console.present(@messages.prompt_for_row)
+  def present_winning_message
+    @console.present(@message.winning_message)
   end
 
-  def user_input_column
-    @console.present(@messages.prompt_for_column)
+  def present_retry_message
+    @console.present(@message.retry_message)
   end
 
-  def present_board(board)
-    present_header_with_numbers(board)
-    board.each do |letter, cells|
-      each_row = cells.map do |cell|
-        reveal(cell)
-      end
-    rows = each_row.join(" ")
-    @console.present("#{letter} #{rows}")
+  def present_hit_boat_message
+    @console.present(@message.hit_message)
+  end
+
+  def present_goodbye_message
+    @console.present(@message.goodbye)
+  end
+
+  def prompt_for_ship_row
+    @console.present(@message.prompt_for_ship_row)
+  end
+
+  def prompt_for_ship_column
+    @console.present(@message.prompt_for_ship_column)
+  end
+
+  def get_valid_ship_row(size_of_board)
+    rows = @validator.possible_rows
+    row_for_ship = user_input_coordinate.upcase
+    if !rows.include?(row_for_ship)
+      prompt_for_ship_row
+      get_valid_ship_row(size_of_board)
+    else
+      row_for_ship
     end
   end
 
-  def present_header_with_numbers(board)
-    columns = (0..board.count).to_a
-    numbers = columns.join(" ")
-    @console.present("#{numbers}")
+  def incorrect_ship_coordinate
+    @console.present(@message.invalid_coordinate)
   end
 
-  def reveal(cell)
-    if cell != @messages.miss_mark && cell != @messages.hit_mark
-      @messages.board_mark
+  def get_valid_ship_column(ship_size)
+    columns = @validator.valid_column(ship_size)
+    column_for_ship = user_input_coordinate.to_i
+    if !columns.include?(column_for_ship)
+      prompt_for_ship_column
+      get_valid_ship_column(ship_size)
     else
-      cell
+      column_for_ship
     end
   end
 
-  def present_winning_message
-    @console.present(@messages.winning_message)
+  def get_right_direction_coordinates(ship_size, board)
```

Comment:
this method should come after it's calling method `get_valid_coordinates`

Reply:
this method has more logic in it than just console related responsibilities.  eg it is maintaining the list of coordinates and it is doing the logic in the `extract_coordinates` method